SalesforceCommerceCloud/pwa-kit
# Feat: Add more e2e tests (@W-16773028@)

# Description

This PR refactors existing e2e tests and adds some new ones for both mobile and desktop:
- Guest shopper can edit product item in cart
- Guest shopper can checkout product bundle
- Registered user can add product to wishlist
- Registered user can see previous orders in order history page



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- (change1)

# How to Test-Drive This PR

1. `git checkout ju/add-e2e-tests` 
2. `npx playwright install`
3. `npm run test:e2e:ui` (or `npm run test:e2e` to see test results in the terminal)
4. Observe success for both desktop and mobile tests

NOTE: If your environment is like mine and does not run the mobile tests, the way I was able to workaround it was editing `playwright.config.js`. I just modified the `projects` like the following:

```js
  projects: [
    {
    // name: "chromium",
    //  use: { ...devices["Desktop Chrome"] },
    //  testIgnore: "**/mobile/**",
    // },

    /* Test against mobile viewports. */
    {
      name: "Mobile Chrome",
      use: { ...devices["Pixel 5"] },
      // testIgnore: "**/desktop/**",
    },
  ],
```

All the tests will now run in mobile, but if you run `npm run test:e2e:ui` you can choose to run only the mobile tests. The CI still runs both desktop and mobile without this modification.


NOTE: As of 10/11/2024, some of the deployments on the e2e environment have seen a lot of `Internal Server Errors`, the clean build I've been working with is commit `886ba9c4fabf00e6fc72de5f63cbe6cac83ce25b` deployment #11711. You may have to redeploy in MRT

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
